hs_modflow_modelinstance/models.py

```diff
@@ -0,0 +1,799 @@
+                         null=True,
+                         blank=True,
+                         verbose_name='Length of transient state stress period(s)')
+
+    def __unicode__(self):
+        return self.stressPeriodType
+
+    class Meta:
+        # StressPeriod element is not repeatable
+        unique_together = ("content_type", "object_id")
+
+    @classmethod
+    def create(cls, **kwargs):
+        kwargs = cls._validate_params(**kwargs)
+        stress_period = super(StressPeriod, cls).create(**kwargs)
+        return stress_period
+
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        kwargs = cls._validate_params(**kwargs)
+        stress_period = super(StressPeriod, cls).update(element_id, **kwargs)
+        delete_if_empty(stress_period)
+
+    @classmethod
+    def _validate_params(cls, **kwargs):
+        for key, val in kwargs.iteritems():
+            if key == 'stressPeriodType':
+                kwargs[key] = validate_choice(val, cls.stressPeriodTypeChoices)
+            elif key == 'transientStateValueType':
+                kwargs[key] = validate_choice(val, cls.transientStateValueTypeChoices)
+        return kwargs
+
+
+class GroundWaterFlow(AbstractMetaDataElement):
+    term = 'GroundWaterFlow'
+    flowPackageChoices = (('BCF6', 'BCF6'), ('LPF', 'LPF'), ('HUF2', 'HUF2'),
+                          ('UPW', 'UPW'), ('HFB6', 'HFB6'), ('UZF', 'UZF'), ('SWI2', 'SWI2'),)
+    flowParameterChoices = (('Hydraulic Conductivity', 'Hydraulic Conductivity'),
+                            ('Transmissivity', 'Transmissivity'),)
+
+    flowPackage = models.CharField(max_length=100, choices=flowPackageChoices, null=True,
+                                   blank=True, verbose_name='Flow package')
+    flowParameter = models.CharField(max_length=100, choices=flowParameterChoices, null=True,
+                                     blank=True, verbose_name='Flow parameter')
+
+    def __unicode__(self):
+        return self.flowPackage
+
+    class Meta:
+        # GroundWaterFlow element is not repeatable
+        unique_together = ("content_type", "object_id")
+
+    @classmethod
+    def create(cls, **kwargs):
+        kwargs = cls._validate_params(**kwargs)
+        gw_flow = super(GroundWaterFlow, cls).create(**kwargs)
+        return gw_flow
+
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        kwargs = cls._validate_params(**kwargs)
+        gw_flow = super(GroundWaterFlow, cls).update(element_id, **kwargs)
+        delete_if_empty(gw_flow)
+
+    @classmethod
+    def _validate_params(cls, **kwargs):
+        for key, val in kwargs.iteritems():
+            if key == 'flowPackage':
+                kwargs[key] = validate_choice(val, cls.flowPackageChoices)
+            elif key == 'flowParameter':
+                kwargs[key] = validate_choice(val, cls.flowParameterChoices)
+        return kwargs
+
+
+class SpecifiedHeadBoundaryPackageChoices(models.Model):

```

Comment:
It would be better to create an Abstract model class (AbstractBoundaryPackageChoices) and then this class and the other 2 similar classes inherit from it.